### PR TITLE
Visualize graphs fix

### PR DESCRIPTION
**Pull request by DexterQvL**
Functional version that visualizes graphs and properly caches.

### CODE REVIEW

File: src/main/java/datastructure/Node.java [hunk 1]
```diff
@@ -135,25 +225,6 @@ public boolean equals(final Object other) {
     }
 
     /**
-     * Converts this Node to a hashcode.
```
Comment (jjtiessen) [inline src/main/java/datastructure/Node.java]: Why is the hashCode removed?

File: src/main/java/parsing/Parser.java [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+            kahnAlgorithm(graph);
+            createCache(absoluteFilePath, graph);
+        } catch (FileNotFoundException e) {
+            System.out.println("Wrong file Destination");
+            e.printStackTrace();
+        } catch (IOException e) {
+            System.out.println("Error while reading file");
+            e.printStackTrace();
+        }
+
```
Comment (jjtiessen) [inline src/main/java/parsing/Parser.java]: Too many blank lines.

File: src/main/java/parsing/Parser.java [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+                writer.newLine();
+                int[] tempList = temp.getOutgoingEdges();
+                size = tempList.length;
+                writer.write("" + size);
+                writer.newLine();
+                for (int j = 0; j < size; j++) {
+                    writer.write("" + tempList[j]);
+                    writer.newLine();
+                }
+
```
Comment (jjtiessen) [inline src/main/java/parsing/Parser.java]: Empty line.

File: src/main/test/classTest.java [hunk 4]
```diff
@@ -1,9 +0,0 @@
-/**
```
Comment (jjtiessen) [inline src/main/test/classTest.java]: How has this ended up in the master...

File: src/main/java/datastructure/Node.java (hunk 1 above)
Comment (DexterQvL) [inline src/main/java/datastructure/Node.java]: Because hashcode was causing hashmaps to malfunction

File: src/main/java/window/Controller.java [hunk 5]
```diff
@@ -1,56 +1,103 @@
...
+     * TextArea used to print information about the running process.
+     */
     @FXML private TextArea console;
 
+    /**
+     * Event handler for when a node or edge is clicked.
+     */
+    private EventHandler<MouseEvent> click = event -> {
+        if (event.getSource() instanceof Rectangle) {
+            DrawNode rect = (DrawNode) (event.getSource());
```
Comment (tvheems) [inline src/main/java/window/Controller.java]: Why is this called "rect"? Please use a more descriptive name.

File: src/main/java/window/Controller.java [hunk 6]
```diff
@@ -59,20 +106,80 @@
...
+     */
+    private void drawGraphUtil(Node center, int radius) {
+        double x = center.getX();
+        for (int i = 0; i < NodeGraph.getCurrentInstance().getSize(); i++) {
+            Node current = NodeGraph.getCurrentInstance().getNode(i);
+            if (current.getX() - 543 > NodeGraph.getCurrentInstance().getMaxX()) {
+                NodeGraph.getCurrentInstance().setMaxX(current.getX() - 543);
+            }
+            if (current.getX() >= x - 40 * radius && current.getX() <= x + 40 * radius) {
+                DrawNode newRect = new DrawNode(i);
```
Comment (tvheems) [inline src/main/java/window/Controller.java]: Another ambiguous name.

File: src/main/java/parsing/Parser.java (hunk 3 above)
Comment (MStraatman) [inline src/main/java/parsing/Parser.java]: Fixed

File: src/main/java/parsing/Parser.java (hunk 2 above)
Comment (MStraatman) [inline src/main/java/parsing/Parser.java]: Fixed

File: src/main/java/datastructure/Node.java (hunk 1 above)
Comment (jjtiessen) [inline src/main/java/datastructure/Node.java]: Okay, that is kinda weird.

File: src/main/java/window/Controller.java (hunk 5 above)
Comment (jjtiessen) [inline src/main/java/window/Controller.java]: Maybe because it is an instance of Rectangle?

**Pull request merged**